src/module.js

```diff
@@ -1,19 +1,51 @@
 /**
  * @constructor
+ * @extends {Sk.builtin.object}
  */
-Sk.builtin.module = function module () {
-};
-Sk.exportSymbol("Sk.builtin.module", Sk.builtin.module);
-
-Sk.builtin.module.prototype.ob$type = Sk.builtin.type.makeIntoTypeObj("module", Sk.builtin.module);
-Sk.builtin.module.prototype.tp$getattr = Sk.builtin.object.prototype.GenericGetAttr;
-Sk.builtin.module.prototype.tp$setattr = Sk.builtin.object.prototype.GenericSetAttr;
-Sk.builtin.module.prototype.tp$name = "module";
+Sk.builtin.module = Sk.abstr.buildNativeClass("module", {
```

Comment:
The api?

Reply:
The fact that a module is a real thing now not a completely empty class.

Reply:
Ah yes and it leads nicely to dealing with $d in the constructor as opposed to in the import maybe. 

I think there’s another pr I remember seeing somewhere that looked at this. I think it was in #991

Surprisingly module is technically subclassable. Though I don’t know how that would be done since it’s not in python’s `__builtins__`

Reply:
I think this breaks down when you `setattr` a `__getattr__` onto a newly created module object. My library does this for mocking purposes, creating fake versions of other modules with special `__getattr__` functions that return data. I'm not clear what the proper solution is - it seems like the `tp$getattr` would need to check to see if `$d` has a `__getattr__` field, and then call that function instead - but my own tests with that don't work for cases where the returned field is a function. Any insight about this?

Reply:
Okay, turns out some other code had a bug; the code below actually seems to work. However, I'm pretty sure it's not at all a good way to handle this. How should this correctly check the existence of a custom __getattr__ that is given through `setattr`?

```javascript
let customGetAttr = this.$d["__getattr__"];
if (customGetAttr) {
    const ret = Sk.misceval.callsimArray(customGetAttr, [pyName]);
    if (ret !== undefined) {
        return ret;
    }
}
```

Reply:
here's the pep for this right? https://www.python.org/dev/peps/pep-0562/

I think you're code looks good to me... so long as it happens after failing to find an `object` in the `module` namespace in line with the pep... It probably needs some suspension awareness thrown in since it seems the perfect place to lazy load something and make a call to the server... which is one of the intended use cases from the pep...

in this pr...
`setattr` will call `Sk.gener.setAttr/object.prototype.tp$setattr` 
so in this case `__getattr__` should be added to `$d` on the module object... and then later called by `module.prototype.tp$getattr` in the expected way... I expect?

_(nb: I have intend to customise the `tp$setattr` for `modules` since in this pr it becomes the only `skulpt object` where `$d` is a javascript object and not a `skulpt dict` so seems sensible to have a custom version)_

p.s. if you find bugs shout so I can fix them!
p.p.s. please don't rely on this pr! My current approach is to make incremental changes that are non breaking - until they're exhausted (or I am 😉 )... I doubt this pr will ever get merged in it's current form...